### ISSUE TITLE
Solve square problem to initialize regression problem using parmest using `objective_at_theta()`

### ISSUE THREAD

**Issue by kanishka-ghosh** [Pyomo/pyomo#2377]
## Summary

Updating the function `objective_at_theta()` to support the option `theta_values=None` instead of making changes to `theta_est()` to solve a square instance of the parameter estimation problem in `parmest` for complex Pyomo models before solving the actual regression problem.  @adowling2 @blnicho @jsiirola this issue supercedes PR #2322.  I will close PR #2322 and open a new PR once this is implemented.

### Rationale

Instead of modifying `theta_est()`, use existing functionality from `objective_at_theta()` to solve initial square problem.

### Description

Following are the steps for the proposed functionality:

1. Update `objective_at_theta()` to handle `theta_values = None`: Parameters are fixed at initialized values and model is solved as a square problem 
2. Unfix parameters before `theta_est()` is called for parameter estimation
3. Add documentation explaining the sequence of solving a parameter estimation problem with an initial square solve on [this page](https://pyomo.readthedocs.io/en/stable/contributed_packages/parmest/driver.html).
    a. Create Estimator objects
    b. Call `objective_at_theta(theta_values=None)`
    c. (Optional) Specify `IPOPT` solver options to initialize parameter estimation from the square problem solution (`bound_push`)
    d. Call `theta_est()`

**Comment by adowling2** [comment]
And to remind everyone, our overarching goal is for ParmEst to model systematically initialize the model using a provided value of theta.

The proposal here is to use `objective_at_theta()` to solve the square problem for each scenario (data set) independently, then copy the initialized values into the "full" (2-stage stochastic program) instance of the parameter estimation model (which is created and solved by `theta_est()` currently).

**Comment by kanishka-ghosh** [comment]
Created a PR (#2438 ) to merge the changes suggested in this issue into main.  However, the codecov tests through GitHub actions are failing repeatedly with the following error for all 3 platforms (linux, win, osx):
<img width="729" alt="image" src="https://user-images.githubusercontent.com/63417433/174866799-a0330a7f-7b3d-4170-8d13-57c264ec4fb2.png">

While searching for solutions to fix this, I found the following GitHub issue: https://github.com/codecov/codecov-action/issues/598 which was closed using the PR: https://github.com/CCExtractor/sample-platform/pull/667

The solution seems to be to downgrade codecov github action to use v0.1.15.

@blnicho It would be great if we could discuss this during today's (06/21/2022) dev call.
